src/odemis/driver/technolution.py

```diff
@@ -307,77 +384,78 @@ def checkCharacters(input, allowed_characters):
+        elif re.search(ASM_HOST_ILLEGAL_CHARS, url_parser.hostname):
+            raise ValueError(
                     "Host contains invalid characters, host remains unchanged "
-                    "(only the characters '%s' are allowed)" % ASM_HOST_ALLOWED_CHARS[2:-1])
-            return self.externalStorageURL.value
+                    "(only the characters '%s' are allowed)." % ASM_HOST_ILLEGAL_CHARS[2:-1])
 
-        elif not checkCharacters(url_parser.path, ASM_PATH_ALLOWED_CHARS):
-            logging.warning("Path on ftp server contains invalid characters, path remains unchanged "
-                            "(only the characters '%s' are allowed)" % ASM_PATH_ALLOWED_CHARS[2:-1])
+        elif re.search(ASM_PATH_ILLEGAL_CHARS, url_parser.path):
+            raise ValueError("Path on ftp server contains invalid characters, path remains unchanged "
+                             "(only the characters '%s' are allowed)." % ASM_PATH_ILLEGAL_CHARS[2:-1])
         else:
-            return url_parser
+            return url
 
 
 class EBeamScanner(model.Emitter):
     """
-    Represents the e-beam scanner of a single field image.
+    HW component representing the e-beam scanner.
     """
 
     def __init__(self, name, role, parent, **kwargs):
         """
-        Initialize the e-beam scanner of a single field image..
+        Initialize the e-beam scanner.
 
-        :param name:
-        :param role:
-        :param parent:
-        :param kwargs:
+        :param name(str): Name of the component
+        :param role(str): Role of the component
+        :param parent (AcquisitionServer object): Parent object of the component
         """
         super(EBeamScanner, self).__init__(name, role, parent=parent, **kwargs)
 
         clockFrequencyData = self.parent.ASM_API_Get_Call("/scan/clock_frequency", 200)
-        # Check if clockFrequencyData holds the proper keysa
-        if 'frequency' not in clockFrequencyData:
-            raise IOError("Could not obtain clock frequency, received data does not hold the proper key")
         self.clockPeriod = model.FloatVA(1 / clockFrequencyData['frequency'], unit='s', readonly=True)
-        self._shape = (6400, 6400)
-        # The resolution min/maximum are derived from the effective cell size restriction defined in the API
+        self._shape = (8000, 8000)
         self.resolution = model.ResolutionVA((6400, 6400), ((10, 10), (1000 * 8, 1000 * 8)))
         self.dwellTime = model.FloatContinuous(self.clockPeriod.value, (max(self.clockPeriod.value, 4e-7), 1e-4),
                                                unit='s')
         self.pixelSize = model.TupleContinuous((4e-9, 4e-9), range=((1e-9, 1e-9), (1e-3, 1e-3)), unit='m',
                                                setter=self._setPixelSize)
-        self.rotation = model.FloatContinuous(0, range=(0, 2 * numpy.pi), unit='rad')
+        self.rotation = model.FloatContinuous(0.0, range=(0.0, 2 * math.pi), unit='rad')
```

Comment:
Just for info, it's always fine to write an int in a FloatContinuous. Integers are a subtype of Float (mathematically speaking).